sdk/cosmos/src/errors.rs

```diff
@@ -1,45 +1,51 @@
-/// A general error having to do with Cosmos.
-#[allow(missing_docs)]
+/// An error having to do with Cosmos.
 #[non_exhaustive]
 #[derive(Debug, thiserror::Error)]
 pub enum Error {
+    /// An error as defined in the `azure_core` crate
     #[error(transparent)]
     AzureCoreError(#[from] azure_core::Error),
+    /// An error related to parsing
     #[error(transparent)]
     ParsingError(ParsingError),
-    #[error("Header not found: {0}")]
-    HeaderNotFound(String),
-    #[error("To str error: {0}")]
-    ToStrError(#[from] http::header::ToStrError),
-    #[error("http error: {0}")]
-    AzureHttpError(#[from] azure_core::HttpError),
-    #[error("stream error: {0}")]
-    StreamError(#[from] azure_core::StreamError),
+    // TODO forward this to azure_core::Error
+    #[allow(missing_docs)]
     #[error("http error: {0}")]
     HttpError(#[from] http::Error),
-    #[error("JSON error: {0}")]
-    JsonError(#[from] serde_json::Error),
-    #[error("UTF-8 conversion error: {0}")]
-    Utf8Error(#[from] std::str::Utf8Error),
-    #[error("base64 decode error: {0}")]
-    DecodeError(#[from] base64::DecodeError),
     /// Other errors that can happen but are unlikely to be matched against
     #[error(transparent)]
     Other(#[from] Box<dyn std::error::Error + Send + Sync>),
 }
 
+/// A parsing error
+///
+/// Most issues are already defined in `azure_core`
 #[derive(Debug, thiserror::Error)]
 pub enum ParsingError {
-    #[error("Parse int error: {0}")]
-    ParseIntError(#[from] std::num::ParseIntError),
-    #[error("uuid error: {0}")]
-    ParseUuidError(#[from] uuid::Error),
-    #[error("Date time parse error: {0}")]
-    ParseDateTimeError(#[from] chrono::format::ParseError),
-    #[error("Parse float error: {0}")]
-    ParseFloatError(#[from] std::num::ParseFloatError),
     #[error("Resource quota parsing error: {0}")]
     ParseResourceQuotaError(#[from] crate::resource_quota::ResourceQuotaParsingError),
-    #[error("parsing error: {0}")]
-    Other(#[from] azure_core::ParsingError),
+    #[error(transparent)]
+    Other(azure_core::ParsingError),
+}
+
+impl<T: Into<azure_core::ParsingError>> From<T> for ParsingError {
+    fn from(error: T) -> Self {
+        Self::Other(error.into())
+    }
+}
+
+impl From<serde_json::Error> for Error {
+    fn from(error: serde_json::Error) -> Self {
+        Self::AzureCoreError(azure_core::Error::JsonError(error))
```

Comment:
That was not my intention with this. I collapsed most of the errors so that in general there's only one error variant for each logical error case. JSON errors are something that impact _all_ SDKs so it belongs in the Core crate even if the error occurs in code that lives in the cosmos crate. Only errors that are specific to/can only happen in cosmos will live inside of cosmos.

Reply:
I'd prefer to discuss this in an issue: Should cosmos_crate be able to create azure_core errors?